### ISSUE TITLE
Number of occurrences

### ISSUE THREAD

**Issue by kdelmonte** [chilts/node-date-recur#2]
Could you add functionality to pass the number of occurrences to `end()`?

**Comment by chilts** [comment]
Good idea. I think an extra method `.for(6)` would be better than overloading `.end()`. ie. for 6 occurrences. Does that sound okay? Note: it would require that a `.start()` date has been set.

Implementation note: I think `.for()` would loop from the `.start()` and check against every date until it hit the nth time, at which point it would internally set the `.end()` date itself.

e.g.

``` js
recur('2012-01-02')
    .setDayOfWeek(4)
    .for(6)
```

This would internally set 9th Feb 2012 to be the `.end()` date.

**Comment by kdelmonte** [comment]
Sounds perfect. Would for() return the amount of occurrences or would it set another property internally to allow the user to continue chaining?

**Comment by kdelmonte** [comment]
@chilts, so the algorithm is just to loop through every single calendar date until you hit the nth occurrence? If that's the case, I can send you a pull request for this one. Let me know if this is okay.

**Comment by chilts** [comment]
Hi @kdelmonte - yep, the `.for()` would just set the `.end()` date internally and return itself so we can continue chaining. It would just try the date once `.matches()` is called much like any other.

Yeah, you could just loop through everything until the nth one is hit, set the `.end()` and return itself. I guess we _could_ cache this but I think we probably shouldn't so it works just like all of the other methods. Does that sound okay?

I think perhaps the docs in the README should state exactly what this does because I could see that doing 6 occurrences over the next 6 weeks would be fine (an evening course perhaps), trying to loop through all the dates looking for the next 20 leap years probably isn't what someone should do. :)

**Comment by kdelmonte** [comment]
This has been resolved on my branch. please see pull request #3 for more.
